src/vm/runners/cairo_runner.rs

```diff
@@ -519,7 +519,11 @@ impl CairoRunner {
         let hint_data_dictionary = self.get_hint_data_dictionary(&references, hint_processor)?;
         #[cfg(feature = "hooks")]
         vm.execute_before_first_step(self, &hint_data_dictionary)?;
-        while vm.run_context.pc != address {
+        loop {
+            match vm.final_pc_update {
+                Some(pc) if pc == address => break,
+                _ => {}
+            };
```

Comment:
I think this looks simpler:
```suggestion
        while Some(address) != vm.final_pc_update {
```
A `!matches()` would work as well.